src/oc_erchef/apps/chef_objects/test/chef_user_tests.erl

```diff
@@ -287,6 +287,33 @@ parse_binary_json_tests(Version) ->
                                                          create, undefined))
                 || Bad <- BadKeys ]
       end}
+    ]
+    ++
+    [
+     {?VD(lists:flatten(io_lib:format("Errors when invalid ~s", [Field]))),
+      fun() ->
+              HtmlyValue = <<"I <3 Chef">>,
+              UserEJson = {make_min_valid_create_user_ejson()},
+              UserEJson1 = ej:set({Field}, UserEJson, HtmlyValue),
+              ?assertThrow(#ej_invalid{key = Field},
+                           chef_user:parse_binary_json(Version, chef_json:encode(UserEJson1), create, undefined))
+      end
+     }
+     || Field <- [<<"display_name">>, <<"firstname">>, <<"middlename">>, <<"lastname">>]
+    ]
+    ++
+    [
+     {?VD(lists:flatten(io_lib:format("Works with non-ASCII ~s", [Field]))),
+      fun() ->
+              %% "Maryam", #1 female name in the arab world as of 2015
+              Value = <<"مريم 1. O'Mara">>,
```

Comment:
Interesting. Must have to do with my editor not knowing right-to-left text, because it looks different there:
<img width="342" alt="screen shot 2017-06-06 at 17 26 05" src="https://user-images.githubusercontent.com/870638/26837281-5a4e0d84-4add-11e7-9fa0-2475dd363dc9.png">